core/src/main/java/discord4j/core/spec/MessageCreateSpecGenerator.java

```diff
@@ -82,6 +83,12 @@ default List<MessageCreateFields.FileSpoiler> fileSpoilers() {
 
     @Override
     default MultipartRequest<MessageCreateRequest> asRequest() {
+        List<Message.Flag> flagsToApply = new ArrayList<>(this.flags().toOptional().orElse(new ArrayList<>()));
+        if (!flagsToApply.contains(Message.Flag.IS_COMPONENTS_V2) && !this.components().isAbsent() && this.components().get().stream().anyMatch(topLevelComponent -> topLevelComponent.getType().isRequiredFlag())) {
+            flagsToApply.add(Message.Flag.IS_COMPONENTS_V2);
+        }
+        Possible<List<Message.Flag>> pFlagsToApply = Possible.of(flagsToApply);
```

Comment:
I feel like this should be extracted to a utility method. There are now ~5 occurrences of this same (or very similar) code.

Reply:
yeah the thing is "where" is a good place for that...

Reply:
also its not the same create/edit because edit is a Possible Optional and create use Possible